Jetcaster/wear/src/main/java/com/example/jetcaster/ui/player/PlayerScreen.kt

```diff
@@ -125,61 +149,137 @@ private fun PlayerScreen(
             // return a null episode
             val episode = state.playerState.episodePlayerState.currentEpisode
 
-            PlayerScreen(
-                mediaDisplay = {
-                    if (episode != null && episode.title.isNotEmpty()) {
-                        TextMediaDisplay(
-                            title = episode.podcastName,
-                            subtitle = episode.title,
+            val exoPlayer = rememberPlayer(context)
+
+            DisposableEffect(exoPlayer, episode) {
+                episode?.mediaUrls?.let { exoPlayer.setMediaItems(it.map { MediaItem.fromUri(it) }) }
+                val mediaSession = MediaSession.Builder(context, exoPlayer).build()
+
+                exoPlayer.prepare()
+
+                onDispose {
+                    mediaSession.release()
+                    exoPlayer.release()
+                }
+            }
+            Box {
+                PlayerSurface(
+                    player = exoPlayer,
+                    modifier = Modifier.resizeWithContentScale(
+                        contentScale = ContentScale.Fit,
+                        sourceSizeDp = null,
+                    ),
+                )
+                PlayerScreen(
+                    mediaDisplay = {
+                        if (episode != null && episode.title.isNotEmpty()) {
+                            MarqueeTextMediaDisplay(
+                                title = episode.title,
+                                artist = episode.podcastName,
+                                titleIcon = DrawableResPaintable(R.drawable.ic_logo),
+                            )
+                        } else {
+                            TextMediaDisplay(
+                                title = stringResource(R.string.nothing_playing),
+                                subtitle = "",
+                                titleIcon = DrawableResPaintable(R.drawable.ic_logo),
+                            )
+                        }
+                    },
+
+                    controlButtons = {
+                        PodcastControlButtons(
+                            onPlayButtonClick = (
+                                {
+                                    playerScreenViewModel.onPlay()
+                                    exoPlayer.play()
```

Comment:
can these exoplayer methods be called from the viewmodel

Reply:
not sure, this is part of the other CL which I mentioned I didn't authored